src/sysctl.cr

```diff
@@ -0,0 +1,87 @@
+lib LibC
```

Comment:
I second that anything LibC should be in the arch specific under `src/lib_c` with file namings following the actual C headers; thus way we may be capable in the future to generate these on-the-flt.

We prefer POSIX, but don't enforce it, if it requires some specific but public API for a platform, or the function is so common across platforms that it can be relied on, you can use it.

That being said, if something is deprecated please don't use it; it will likely break some day.